analyzer/tools/build-logger/src/ldlogger-hooks.c

```diff
@@ -17,23 +17,51 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
+#include <spawn.h>
 #include <stdbool.h>
 
 #include "ldlogger-hooks.h"
 
+// We are done intercepting system calls by pretending that the functions below
+// are the real thing -- time to call the actual system call.
+#define CALL_ORIGINAL_FN(funName_, arglist, ...) \
+  typedef int (*FunType) arglist; \
+  FunType fun = (FunType) dlsym(RTLD_NEXT, #funName_); \
+  if (!fun) \
+  { \
+    return -1; \
+  } \
+  return (*fun)( __VA_ARGS__ );
+
 #define CC_LOGGER_CALL_EXEC(funName_, arglist, ...) \
   tryLog(__VA_ARGS__); \
   { \
     unsetLDPRELOAD(__VA_ARGS__); \
```

Comment:
Are you confident, that this unsetLDPRELOAD is not needed in the newly implemented spawn wrapper?